src/PolysemyContrib.hs

```diff
@@ -17,6 +19,17 @@ data FileProvider m a where
 
 makeSem ''FileProvider
 
+type InMemFileSystem = Map FilePath T.Text
+
+runFileProviderPure :: (Member (State InMemFileSystem) r) => Sem (FileProvider ': r) a -> Sem r a
+runFileProviderPure = interpret $ \case
+  ReadFile path -> do
+    m <- get @InMemFileSystem
```

Comment:
Does `@InMemFileSystem` come from `Polysemy.State`?

Reply:
No, it's defined few lines above `type InMemFileSystem = Map FilePath T.Text`